chemprop/v2/cli/train.py

```diff
@@ -252,10 +252,9 @@ def add_train_args(parser: ArgumentParser) -> ArgumentParser:
         "-t",
         "--task-type",
         default="regression",
-        action=RegistryAction(
```

Comment:
@davidegraff can you remind me why we use `action=RegistryAction(<registry>)` instead of `choices=<registry>.keys()`? Reading the source code, to me they look equivalent and Xiaorui's change here makes it more simple.

Reply:
the `RegistryAction` sets the `choices` argument automatically, but more importantly it essentially functions as a dictionary lookup. That is when a client calls `--task-type regression`, then `args.task_type` is automatically set to `ReadoutRegistry["regression"]` (which is a `RegressionFFN`).
perhaps (in another PR) we will want to change the `--task-type` `metavar` to something like `metavar="ffn"`

Reply:
yeah wait @xiaoruiDong why is this code changed? The `RegistryAction` should take care of both of these things.

Reply:
1. Sorry that I mistakenly changed this to `ReadoutRegistry.keys()`... I found, in a recent merged PR, Hao-Wei did a similar change and thought this was the preferred way to do.
2. If I read @davidegraff correctly, do we expect to see `args.task_type == RegressionFFN` when someone do `--task-type regression`? Currently the actual behavior is that `args.task_type` is a str `regression` with `action=RegistryAction(ReadoutRegistry)`

Reply:
The second point was mistaken. I had it that way in an old version and I decided to change it back in a previous commit, but maybe we should change it back to avoid some boilerplate client code? In any case, you should be using the `RegistryAction`.